### PR TITLE
Chain configs

### PR DESCRIPTION
**Pull request by coa-telos**
# Fixes #issue_number

## Checklist:

-   [ ] I have performed a self-review of my own code
-   [ ] I have commented my code, particularly in hard-to-understand areas
-   [ ] I have cleaned up the code in the areas my change touches
-   [ ] My changes generate no new warnings
-   [ ] Any dependent changes have been merged and published in downstream modules
-   [ ] I have checked my code and corrected any misspellings
-   [ ] I have removed any unnecessary console messages
-   [ ] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [ ] I have tested for mobile functionality and responsiveness
-   [ ] I have added appropriate test coverage
-   [ ] I have updated relevant documentation and/or opened a separate issue to cover the updates (Issue URL: )

### CODE REVIEW

File: bin/reth/src/cli/mod.rs [hunk 1]
```diff
@@ -271,6 +271,9 @@ mod tests {
     /// Tests that the log directory is parsed correctly. It's always tied to the specific chain's
     /// name
     #[test]
+    #[cfg(not(feature = "telos"))]
```
Comment (poplexity) [inline bin/reth/src/cli/mod.rs:274]: I don't think we need to disable the upstream tests individually like this, there will be a lot of tests that will not work with our changes and we want to make as few changes as possible (to reduce merge conflicts).  I have removed the .github directory and added our own CI workflow which will just run our E2E test.

File: bin/reth/src/optimism.rs [hunk 2]
```diff
@@ -14,7 +14,7 @@ use std::sync::Arc;
 #[global_allocator]
 static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
 
-#[cfg(not(feature = "optimism"))]
+#[cfg(all(not(feature = "optimism"), not(feature = "telos")))]
```
Comment (poplexity) [inline bin/reth/src/optimism.rs:17]: I see you also removed a similar change from `bin/reth/src/main.rs`, do we need either of these?  If we always build with the `telos` feature flag, should we need to worry at all about the other binaries besides our own `bin/reth/src/telos.rs`?  If possible, neither of these changes would be ideal, so we have fewer changes/merge conflicts.

File: crates/chainspec/src/spec.rs [hunk 3]
```diff
@@ -30,6 +30,8 @@ use reth_network_peers::{
     base_nodes, base_testnet_nodes, holesky_nodes, mainnet_nodes, op_nodes, op_testnet_nodes,
     sepolia_nodes,
 };
+#[cfg(feature = "telos")]
```
Comment (poplexity) [inline crates/chainspec/src/spec.rs:33]: Initially we will not have the reth nodes doing any peering, the Telos node itself peers on the network and then reth is just consuming the blocks from that node.  Possibly after do our initial release we will want to find a way to run reth without a Telos node, which could use peering, but we haven't planned for that yet.

So, if this is needed for reth to function then that's good, but if this is only needed to allow reth to peer with other reth nodes, we can skip this change for the initial release.

Comment (poplexity) [reply crates/chainspec/src/spec.rs:33]: This would apply to all the other changes related to peers and bootnodes.

**Review by poplexity — COMMENTED** [review]
Looks good to me, left some comments that are generally focused on reducing the number of changes we make so we can reduce the chances of conflicts during future merges.

File: bin/reth/src/optimism.rs (hunk 2 above)
Comment (coa-telos) [reply bin/reth/src/optimism.rs:17]: not sure I'm following, you think that its OK to make it compilable only with our feture flag, and doing that way will reduce differences?

**Pull request merged**